### PR TITLE
[Client/Crash] Fix exploit spam duel starting.

### PR DESCRIPTION
**Pull request by masterking32**
Check this issue: [https://github.com/TrinityCore/TrinityCore/issues/22374](https://github.com/TrinityCore/TrinityCore/issues/22374)

### How to test:
[https://github.com/tomrus88/WowAddin](https://github.com/tomrus88/WowAddin)
and use this script:
```
BOOL DuelAccept(char const*, char const* String)
{
	long Count = atoi(String);

	if (Count <= 0)
		Count += 1;

	for (long A = 0; A < Count; ++A)
	{
		if (Delay > 1)
			Sleep(Delay);

		CDataStore Data(CMSG_DUEL_ACCEPTED);
		Data.PutInt64(GetTargetGuid());
		Data.Finalize();
		ClientServices::SendPacket(&Data);
	}

	Console::Write("Packet Sent %u Times", ECHO_COLOR, Count);
	return true;
}
```

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
can't test on linux

@IntelligentQuantum have you tested before and after this PR?

**Comment by IntelligentQuantum** [comment]
@BarbzYHOOL Yeah, it works fine.

**Comment by BarbzYHOOL** [comment]
TC commit is huge tho https://github.com/TrinityCore/TrinityCore/commit/003cc56d70c9b07f4484cb82484091ab4013895a

**Comment by IntelligentQuantum** [comment]
We have 1000 way to fix one issue xD

**Comment by BarbzYHOOL** [comment]
need the best way

File: src/server/game/Entities/Player/Player.h [hunk 1]
```diff
@@ -282,14 +282,15 @@ struct PvPInfo
...
-    DuelInfo() : initiator(NULL), opponent(NULL), startTimer(0), startTime(0), outOfBound(0), isMounted(false) {}
+    DuelInfo() : initiator(NULL), opponent(NULL), startTimer(0), startTime(0), outOfBound(0), isMounted(false), underDuel(false) {}
 
     Player* initiator;
     Player* opponent;
     time_t startTimer;
     time_t startTime;
     time_t outOfBound;
     bool isMounted;
+	bool underDuel;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.h]: please fix indentation

File: src/server/game/Handlers/DuelHandler.cpp [hunk 2]
```diff
@@ -35,9 +35,16 @@ void WorldSession::HandleDuelAcceptedOpcode(WorldPacket& recvPacket)
     sLog->outStaticDebug("Player 2 is: %u (%s)", plTarget->GetGUIDLow(), plTarget->GetName().c_str());
 #endif
 
+	if(player->duel->underDuel || plTarget->duel->underDuel)
+	{
+		return;
+	}
+
```
Comment (FrancescoBorzi) [inline src/server/game/Handlers/DuelHandler.cpp]: please fix indentation

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -7801,6 +7801,9 @@ void Player::DuelComplete(DuelCompleteType type)
     sLog->outDebug(LOG_FILTER_UNITS, "Duel Complete %s %s", GetName().c_str(), duel->opponent->GetName().c_str());
 #endif
 
+	duel->initiator->duel->underDuel = false;
+	duel->opponent->duel->underDuel = false;
+
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.cpp]: please fix indentation

**Comment by Viste** [comment]
> TC commit is huge tho

they cleaned and reorganized the code. masterking32 put a crutch in the existing code. it's okay.

File: src/server/game/Handlers/DuelHandler.cpp (hunk 2 above)
Comment (masterking32) [reply src/server/game/Handlers/DuelHandler.cpp]: They are ok in my file, I use notepad++ xD and when you open the file "View file" in Github that's ok. I cannot solve that with my IDE can u do it? :D

Comment (Viste) [reply src/server/game/Handlers/DuelHandler.cpp]: switch your notepad from tabs to 4 space )

Comment (Viste) [reply src/server/game/Handlers/DuelHandler.cpp]: or give me acces to your repo i fix it for you

**Comment by FrancescoBorzi** [comment]
@masterking32 it looks like the PR's branch/repository has been deleted :(

File: src/server/game/Handlers/DuelHandler.cpp (hunk 2 above)
Comment (masterking32) [reply src/server/game/Handlers/DuelHandler.cpp]: Repo already removed =)))

**Comment by masterking32** [comment]
Yes :( sorry...

**Pull request closed without merging**